runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunnerTest.java

```diff
@@ -253,6 +290,43 @@ public void testRunStreamingJobThatFails() throws Exception {
     fail("AssertionError expected");
   }
 
+  private LowLevelHttpResponse generateMockStreamingMetricResponse(
```

Comment:
The meaning of the parameters and the mapping from those parameters to the watermarks are not obvious to me.

maybe:
generateMockStreamingMetricResponse(Map<String, Long> metrics...)